modules/mobile-agents/windows/jax-rs/src/main/java/org/wso2/carbon/mdm/mobileservices/windows/services/syncml/impl/SyncmlServiceImpl.java

```diff
@@ -0,0 +1,285 @@
+						MOBILE_DEVICE_TYPE_WINDOWS, devID, OSVersion, IMSI, IMEI, devMan, devMod);
+					try {
+						SyncmlUtils.getDeviceManagementService()
+						                     .enrollDevice(generatedDevice);
+					} catch (DeviceManagementException e) {
+						String msg = "Exception while getting Device Management Service.";
+						logger.error(msg, e);
+						throw new DeviceManagementException(msg, e);
+					} catch (DeviceManagementServiceException e) {
+						String msg = "Exception while enrolling device after receiving data.";
+						logger.error(msg, e);
+						throw new DeviceManagementServiceException(msg, e);
+					}
+				}
+			}
+		}
+		String response = prepareResponse(msgID, targetURI, sourceURI);
+		return Response.ok().entity(response).build();
+	}
+
+	/**
+	 * This method is used to generate and return Device object from the received information at
+	 * the Syncml step.
+	 *
+	 * @param deviceID     - Unique device ID received from the Device
+	 * @param OSVersion    - Device OS version
+	 * @param IMSI         - Device IMSI
+	 * @param IMEI         - Device IMEI
+	 * @param manufacturer - Device Manufacturer name
+	 * @param model        - Device Model
+	 * @return - Device Object
+	 */
+	private static Device generateDevice(String type, String deviceID, String OSVersion, String IMSI,
+	                                    String IMEI, String manufacturer, String model) {
+
+		Device generatedDevice = new Device();
+
+		Device.Property OSVersionProperty = new Device.Property();
+		OSVersionProperty.setName(OS_VERSION);
+		OSVersionProperty.setValue(OSVersion);
+
+		Device.Property IMSEIProperty = new Device.Property();
+		IMSEIProperty.setName(SyncmlServiceImpl.IMSI);
+		IMSEIProperty.setValue(IMSI);
+
+		Device.Property IMEIProperty = new Device.Property();
+		IMEIProperty.setName(SyncmlServiceImpl.IMEI);
+		IMEIProperty.setValue(IMEI);
+
+		Device.Property DevManProperty = new Device.Property();
+		DevManProperty.setName(VENDOR);
+		DevManProperty.setValue(manufacturer);
+
+		Device.Property DevModProperty = new Device.Property();
+		DevModProperty.setName(MODEL);
+		DevModProperty.setValue(model);
+
+		List<Device.Property> propertyList = new ArrayList<Device.Property>();
+		propertyList.add(OSVersionProperty);
+		propertyList.add(IMSEIProperty);
+		propertyList.add(IMEIProperty);
+		propertyList.add(DevManProperty);
+		propertyList.add(DevModProperty);
+
+		generatedDevice.setDeviceIdentifier(deviceID);
+		generatedDevice.setProperties(propertyList);
+		generatedDevice.setType(type);
+
+		return generatedDevice;
+	}
+
+	/**
+	 * Method for preparing the SyncML response.
+	 *
+	 * @param msgID - Incoming message ID
+	 * @param targetURI - Target URI in SyncML message
+	 * @param sourceURI - Sourse URI in SyncML message
+	 * @return - Returns the SyncML response as a String
+	 * @throws FileOperationException
+	 */
+	private String prepareResponse(String msgID, String targetURI, String sourceURI)
+			throws FileOperationException {
```

Comment:
Keep the keyword "throws" within the first line itself. When you break the line from that particular point, it gives an indication that there are some exceptions following the method declaration.